packages/jest-transform/src/__tests__/ScriptTransformer.test.ts

```diff
@@ -1989,7 +2026,12 @@ describe('ScriptTransformer', () => {
       getCoverageOptions(),
     );
 
-    expect(fs.readFileSync).toHaveBeenCalledTimes(4);
+    /**
+     * 4 times for fetching the file.
+     * 1 time for fetching the cached transform file created by
+     * a previous transformer that returned the same cache key for the file...
+     */
+    expect(fs.readFileSync).toHaveBeenCalledTimes(5);
```

Comment:
I traced through the logic and the 5th call is `readCodeCacheFile`. I think that means that instead of transforming from scratch, the last transformer's hash matched the hash of a previous transformer's cache file. This might be a bug? Maybe the hash should take the transformer's name into account?. 

Now that I call `getCacheKey` earlier it means we call `writeCodeCacheFile` sooner, so it could also be a race condition? I'm not sure. It might just be because of the mocks returning the same hash for the same file contents?

Reply:
@SimenB I did some more debugging and this looks expected. both `yetAnotherScriptTransformer` and `scriptTransformer` use the same `transformConfig` so they share the same `projectCache`. I'm not sure why the transform output of  `scriptTransformer` wasn't being cached before, but now it is and `yetAnotherScriptTransformer` finds it and reads that existing cache rather than transforming from scratch again. So I think this is an improvement? :)

Reply:
actually. maybe I found a bug. let me put up a PR. I think we need to include the `configString` when making the cached file / directory so that it doesn't collide with a different transformer that happens to return the same `cacheKey`